### PR TITLE
fix(ESSNTL-3336): fix os column sorting

### PR DESCRIPTION
**Pull request by mkholjuraev**
Resolves: [ESSNTL-3336](https://issues.redhat.com/browse/ESSNTL-3336)

Fixes the OS columns sorting by making the column key and sort key consistent system_profile. In case, they are different, sorting does not work and we need to provide operating.

### CODE REVIEW

File: src/components/InventoryTable/helpers.js [hunk 1]
```diff
@@ -31,22 +31,36 @@ export const createRows = (rows = [], columns = [], { actions, expandable, noSys
...
-        ],
-        parent: key * 2,
-        fullWidth: true
-    }]))).filter(Boolean);
+    return flatten(rows.map((oneItem, key) => {
+        //makes the row column key and sort key consistent, otherwise sorting does not work
+        const rowItem = {
+            ...oneItem,
+            // eslint-disable-next-line camelcase
+            operating_system: oneItem.system_profile?.operating_system
```
Comment (gkarat) [inline src/components/InventoryTable/helpers.js:39]: Is it ok that here can be an undefined value? Will it be treated as "Not available" in the table? Just checking

File: src/components/InventoryTable/helpers.js [hunk 2]
```diff
@@ -31,22 +31,36 @@ export const createRows = (rows = [], columns = [], { actions, expandable, noSys
...
-    }, oneItem.children && expandable && {
-        cells: [
-            {
-                title: typeof oneItem.children === 'function' ? oneItem.children() : oneItem.children
-            }
-        ],
-        parent: key * 2,
-        fullWidth: true
-    }]))).filter(Boolean);
+    return flatten(rows.map((oneItem, key) => {
```
Comment (gkarat) [inline src/components/InventoryTable/helpers.js:34]: I think you can use `flatMap` here

**Review by gkarat — APPROVED** [review]
Tried locally, everything seems to be fine 👍🏼 good job

File: src/components/InventoryTable/helpers.js (hunk 1 above)
Comment (mkholjuraev) [reply src/components/InventoryTable/helpers.js:39]: Yes, I have verified it.

Comment (mkholjuraev) [reply src/components/InventoryTable/helpers.js:39]: I believe flatmap is just syntactical sugar. I am not a fan of touching what is working. 

_It is identical to a [map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) followed by a [flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) of depth 1 (arr.map(...args).flat())_  see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)

**Comment by mkholjuraev** [comment]
@gkarat thank you for the review!

File: src/components/InventoryTable/helpers.js (hunk 2 above)
Comment (mkholjuraev) [reply src/components/InventoryTable/helpers.js:34]: I believe flatmap is just syntactical sugar. I am not a fan of touching what is working :). 

_It is identical to a [map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) followed by a [flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) of depth 1 (arr.map(...args).flat())_  see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)

**Pull request merged**